dev/com.ibm.ws.transport.http/src/io/openliberty/http/netty/channel/utils/HeaderValidator.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.http.netty.channel.utils;
+
+import java.util.regex.Pattern;
+import com.ibm.ws.http.channel.internal.HttpChannelConfig;
+
+/**
+ * Processes and validates HTTP header names and values in compliance with 
+ * RFC 7230, "Hypertext Transfer Protocol (HTTP/1.1): Message Syntax 
+ * and Routing"
+ * 
+ * This utility ensures:
+ *      Header names are composed of valid "tchar" characters.
+ *      Header names and values do not exceed a configured maximum field length.
+ *      Control characters (except for valid folding) and non-ASCII characters
+ *          are properly handled.
+ *      When following a folding sequence, CR must be followed by LF, and LF 
+ *          must be followed by a whitespace.
+ *      Trailing CR of LF is disallowed. 
+ * 
+ * The class also normalizes header names (converting them to lowercase) and
+ * trims leading/trailing whitespaces from both names and values.
+ */
+public class HeaderValidator {
+
+    /**
+     * Defines a pattern for valid header names (token characters or "tchars") as specified in 
+     * RFC 7230, Section 3.2.6, "Field Value Components".
+     */
+    private static final Pattern TCHAR_PATTERN = Pattern.compile("^[!#$%&'*+\\-\\.\\^_`|~0-9a-zA-Z]+$");
+    private static final char CR = '\r';
+    private static final char LF = '\n';
+    private static final char TAB = '\t';
+    private static final char SPACE = ' ';
+
+    /** 
+     * Enumerates if the field token being processed is a header name or header value.
+    */
+    public enum FieldType{NAME, VALUE}
+
+    private HeaderValidator() {
+        //Utility Singleton
+    }
+
+
+    /**
+     * Peforms processing of a header field (name or value).
+     * 
+     * This method normalizes a header field:
+     *      Ensures a non-null input if field type is {@link FieldType#NAME}.
+     *      Trims the input if it is non-null.
+     *      Substitutes a null input with an empty string if the field type is
+     *          {@link FieldType#VALUE}.
+     *      Lowercases the token if the field type is {@link FieldType#NAME}
+     * 
+     * @param token a raw header field token; may be {@code null} for values, 
+     *      but not for names.
+     * @param type whether this token is a header name or value
+     * @param config the HTTP configuration object
+     * @return processed and possibly normalized header field, ensuring to 
+     *      comply with the configured validation requirements
+     * @throws IllegalArgumentException if the field is invalid (too long, 
+     *      contains illegal characters, or null name)
+     */
+    public static String process(String token, FieldType type, HttpChannelConfig config){
+
+        if(type == FieldType.NAME && token == null){
+            throw new IllegalArgumentException("Header name must not be null");
```

Comment:
Should this exception message be translated? Does the end user see this?

Reply:
Will be addressed in [#30653](https://github.com/OpenLiberty/open-liberty/issues/30653)